dt: 0.02
geofence: [1.5, 2.0, 2.0, -1.0, -3.0, -0.5] # [x_max, y_max, z_max, x_min, y_min, z_min]
origin: [0.0, 0.0, 0.22] # [x, y, z]
takeoff_pos: [0.0, 0.0, 1.0] # [x, y, z]
takeoff_quat: [0.0, 0.0, 0.0, 1.0] # [x, y, z, w]
takeoff_duration: 10.0
landing_duration: 8.0

return_home_pos: [0.0, 0.0, 1.0] # [x, y, z]
return_home_quat: [0.0, 0.0, 0.0, 1.0] # [x, y, z, w]

send_vel: true
send_thrust: true
send_arm: true

mpc: 
  mass: 6.0
  T: 0.64
  N: 32
  N_mini: 8
  # Q: [12, 16, 12, 10, 0.05, 0.05, 0.05, 0.08, 0.08, 0.08] # [p_ee, q_ee(1 dim cost), v, arm_angle]
  # Q: [16, 16, 16, 10, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05] # [p_ee, q_ee(1 dim cost), v, arm_angle]
  # Q: [48, 48, 48, 32, 32, 32, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.1, 0.1, 0.1, 0.1] # [p_ee(3), ee_rpy_err(3), v(3), base_euler(3), base_ang_vel(3), arm_angle(4)] - Reduced base costs for more agility
  Q: [48, 48, 48, 32, 32, 32, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.06, 0.02, 0.02, 0.01] # [p_ee(3), ee_rpy_err(3), v(3), base_euler(3), base_ang_vel(3), arm_angle(4)] - Reduced base costs for more agility
  R: [0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.0, 0.0, 0.0, 0.0]               # [f(3), torque_xyz(3), arm_ref(4)] - Reduced control effort penalties
  R_delta: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.01, 0.01, 0.01, 0.01]      # Reduced smoothing for more responsive control
  acc2thrust_gain: [5.5, 5.5, 5.5]
  moment_of_inertia: [0.05, 0.05, 0.1] # [I_x, I_y, I_z] kg*m^2 - Roll, Pitch, Yaw
  pos_min: [-100.0, -100.0, -100.0]
  vel_min: [-10.5, -10.5, -10.5]     
  vel_max: [10.5, 10.5, 10.5]
  acc_min: [-20.0, -20.0, -20.0]     
  acc_max: [20.0, 20.0, 20.0]
  joint_min: [0.0, 0.0, 0.0, -360.0] # degree
  joint_max: [130.0, 130.0, 150.0, 360.0] # degree
  joint_vel_min: [-10.0, -15.0, -20.0, -20.0] # degree/s
  joint_vel_max: [10.0, 15.0, 20.0, 20.0] # degree/s
  default_arm_angle: [30.0, 32.0, 0.0, 0.0] # Updated for 4-DOF

  output_filter_gain: [0.6, 0.6, 0.6, 0.6, 0.6, 0.6, 0.6, 0.6, 0.6, 0.6] # Updated for full torque control



disturbance_observer:
  enable: false
  cutoff_freq: 0.08
  acc_max: [0.5, 0.5, 0.5]
  acc_min: [-0.5, -0.5, -0.5]